inc/functions/formatting.php

```diff
@@ -109,15 +109,22 @@ function rocket_remove_url_protocol( $url, $no_dots = false ) {
 }
 
 /**
- * Add HTTP protocol to an url that does not have
+ * Add HTTP protocol to an url that does not have and add host if missing too
  *
  * @since 2.2.1
  *
  * @param string $url The URL to parse.
  * @return string $url The URL with protocol
  */
 function rocket_add_url_protocol( $url ) {
-	if ( strpos( $url, 'http://' ) === false && strpos( $url, 'https://' ) === false ) {
+  
+	$link_data = wp_parse_url( $url );
+
+	if( empty($link_data['host']) ) {
```

Comment:
This is making the function have 2 different goals, which is problematic, and the name becomes misleading.

This should be either done directly in the homepage preload method, or with a new function.

Reply:
Done, the rocket_add_url_protocol() remains unaffected now.